integration-tests/src/lib.rs

```diff
@@ -197,6 +217,9 @@ where
 
         [embedding]
         directory = "../assets/smbert_v0003"
+
+        [request_context]
+        disable_legacy_tenant = false
```

Comment:
🔧 i assume this comes from an older iteration of the code, but since the naming is different now this won't have any effect. i'd also assume that `disabled=false` would become `enabled=true`, right? (i guess it didn't crash in the tests, because the default is set to `enabled=true`)
```suggestion
        [tenants]
        enable_legacy_tenant = true
```

Reply:
It's no a `single_tenant_mode` it's an option which allows having a legacy tenant. If we want in parallel with other tenants (not that we need that but we get that for free).

Furthermore in the long run we likely will use the normal deployment templates we have for multi tenant for single tenant, too (to massively reduce maintenance overhead).  Hence this option isn't just about single tenant, but about "old"  deployments we don't do this way anymore.

Hence `legacy`.

Also given that it must work when deploying to production without changing the config means that if the config option is set it is set to disable the legacy tenant, as the default needs to be having it enabled.

Hence `disable` .

And there is no reason why we should even switch the default of the option, that would just be unnecessary work, so the next time the option is touched is when removing it.

Reply:
> i assume this comes from an older iteration of the code, but since the naming is different now this won't have any effect.

yes it doesn't have any effect, just is explicit.

Reply:
We don't have an old system. The current deployment will continue to live as it is. Each of the current tenant will stay in their own infrastructure, they will run in parallel by definition, and they don't need an id. 
What we need to make that work is avoiding to changing the role if we are in single tenant mode and don't look for an header.